### DISCUSSION TITLE
Speed Limits

### DISCUSSION THREAD

**Discussion by Raycock2** [organicmaps/organicmaps#2301] [Ideas]
Add speed limits to maps and the ability to edit them on the fly.

**Comment by starsep** [comment]
I agree that displaying speed limit would be useful and doable as this data is already used by routing algorithm.

Mapping speed limit isn't trivial:
1. Sensible way to choose segment of a road and possibly split it is needed. I think answering what is speed limit on current road while in car navigation mode is not good enough.
2. Solution would need to follow local laws.
3. There are many cases and UI needs be both relatively easy and complete.

I would advise to check what StreetComplete does first https://github.com/streetcomplete/StreetComplete/tree/master/app/src/main/java/de/westnordost/streetcomplete/quests/max_speed

**Reply by fitojb** [reply]
The main gripe I have with StreetComplete is that it only lets you add new speed limits, but it never surfaces _existing_ ones, so when they change, you can’t use that app to survey the updates.

**Comment by alensiljak** [comment]
These are two distinct features. There's already a request for displaying the speed limit: https://github.com/organicmaps/organicmaps/issues/952
I find this extremely valuable when driving in a city. 

Editing speed limits is a bit complex with OSM. This will sound lame but there is one of this osm-helper apps, an open-source app on f-droid (kill me but I can't remember the name anymore) that has a dashboard for quickly setting the speed limit, and a few other operations. I wonder how they're doing it. The source is obviously available. Hopefully someone knows what app I'm referring to.
For editing, someone should suggest a decent workflow from the driver's perspective. Whatever I could come up with turned out way too impractical for using while driving. And totally impractical if trying with Vespucci in a car at a stop afterwards. 
Perhaps just an OSM note with exact location and an automatic direction information?

**Comment by Himra** [comment]
Hello everyone,

I've been looking for an app like Organic Maps for a long time and I'm only missing one thing for my absolute happiness, namely the display of the maximum speed while navigating by car.

Is there already something planned for this?

Thanks for the great app

greetings Himra

**Comment by woelfchen72** [comment]
Hello,

showing the actual speed limit is one of two features I'm missing in organic maps. I would love it in navigation mode, but it would be also nice to have in free drive mode. And in Android Auto Display too.

Thank you for this app. If I get it compiled, maybe I could help you somehow.

**Comment by trymeouteh** [comment]
Would like to see speed limit displayed as well as current speed when driving during navigation

**Comment by PinguDEV-original** [comment]
Well, this feature now exists https://github.com/organicmaps/organicmaps/releases/tag/2025.01.26-9-android :tada:!

**Reply by alensiljak** [reply]
> Well, this feature now exists https://github.com/organicmaps/organicmaps/releases/tag/2025.01.26-9-android 🎉!

At least the first 50% of it.

**Comment by bargais** [comment]
Unfortunately no speed limit information in android auto. 😕
